internal/engine/image_build_and_deployer.go

```diff
@@ -346,10 +355,12 @@ func (ibd *ImageBuildAndDeployer) createEntitiesToDeploy(ctx context.Context,
 					var sidecarInjected bool
 					e, sidecarInjected, err = sidecar.InjectSyncletSidecar(e, injectedRefSelector, ibd.syncletContainer)
 					if err != nil {
-						return nil, err
+						return result, err
```

Comment:
to be clear, we're gracefully handling the "we tried to inject the synclet but didn't" case, but not the "something went WRONG while injecting the synclet" case, right?

Reply:
ya, i renamed the fields a bit to be more specific. We're trying to handle the case where:
1) We injected an image successfully, but
2) We weren't able to find a place to inject the synclet sidecar